### PR TITLE
fix new message button focus issue in chat window

### PR DESCRIPTION
**Pull request by deepalic**
> Fixes #
https://msazure.visualstudio.com/CCI/_workitems/edit/28081591
https://msazure.visualstudio.com/CCI/_workitems/edit/27168818

## Changelog Entry
 Fixes (https://msazure.visualstudio.com/CCI/_workitems/edit/28081591), (https://msazure.visualstudio.com/CCI/_workitems/edit/27168818) in PR (https://github.com/microsoft/BotFramework-WebChat/pull/5207), by [@deepalic](https://github.com/deepalic)

## Description
Added fix to move focus on 'New messages' button in chant window.
Tested the change with NVDA.

-  [ ] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [x] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js [hunk 1]
```diff
@@ -14,6 +14,14 @@ const ScrollToEndButton = ({ onClick }) => {
   const localize = useLocalizer();
 
   const text = localize(scrollToEndButtonBehavior === 'any' ? 'TRANSCRIPT_MORE_MESSAGES' : 'TRANSCRIPT_NEW_MESSAGES');
+  // useRef and useEffect is added to move focus on 'New messages' button when it is display on screen.
+  const buttonReference = useRef();
+
+  useEffect(() => {
+    if (buttonReference?.current) {
+      buttonReference?.current?.focus();
```
Comment (sounakpal) [inline packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js:22]: if (buttonReference?.current) {
      buttonReference.current.focus();
      
      //since you are already checking buttonRef and current in the above line

Comment (deepalic) [reply packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js]: fixed

File: __tests__/html/scrollToEndButton.tabOrder.html [hunk 2]
```diff
@@ -51,9 +51,7 @@
 
         await pageConditions.allImagesLoaded();
 
-        await host.sendTab();
-
-        // After pressing TAB from transcript, it should focus on the "New messages" button.
+        // When "New message" appears then focus should be automatically move to "New messages" button.
```
Comment (OEvgeny) [inline __tests__/html/scrollToEndButton.tabOrder.html:54]: Please revert the change and add a dedicated test covering changes made in the PR

File: packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js [hunk 3]
```diff
@@ -14,6 +14,14 @@ const ScrollToEndButton = ({ onClick }) => {
...
 
   const text = localize(scrollToEndButtonBehavior === 'any' ? 'TRANSCRIPT_MORE_MESSAGES' : 'TRANSCRIPT_NEW_MESSAGES');
+  // useRef and useEffect is added to move focus on 'New messages' button when it is display on screen.
+  const buttonReference = useRef();
+
+  useEffect(() => {
+    if (buttonReference?.current) {
+      buttonReference.current.focus();
+    }
+  }, [buttonReference]);
```
Comment (OEvgeny) [inline packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js:24]:
```suggestion
  const buttonReference = useRef<HTMLButtonElement>();
  
  // Move focus to 'New messages' button when it gets displayed on screen
  useEffect(() => {
    buttonReference.current?.focus();
  }, [buttonReference]);
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ Notes: web developers are advised to use [`~` (tilde range)](https://github.com/
 ## [Unreleased]
 
 ### Fixed
-
+- Fixes (https://msazure.visualstudio.com/CCI/_workitems/edit/28081591), (https://msazure.visualstudio.com/CCI/_workitems/edit/27168818) in PR (https://github.com/microsoft/BotFramework-WebChat/pull/5207), by [@deepalic](https://github.com/deepalic)
```
Comment (compulim) [inline CHANGELOG.md:28]: Please explain what is being fixed.

File: __tests__/html/scrollToEndButton.tabOrder.html (hunk 2 above)
Comment (compulim) [reply __tests__/html/scrollToEndButton.tabOrder.html:54]: We cannot move the focus when "New messages" button appear. This is violating accessibility compliance.

This is because the message come asynchronously and moving the focus will interrupt the screen reader.

**Review by compulim — CHANGES_REQUESTED** [review]
We must not move the focus when message arrive. This will interrupt screen reader and break accessibility compliance.

File: __tests__/html/scrollToEndButton.tabOrder.html (hunk 2 above)
Comment (deepalic) [reply __tests__/html/scrollToEndButton.tabOrder.html]: Fixed in latest commit

File: packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js (hunk 3 above)
Comment (deepalic) [reply packages/component/src/Middleware/ScrollToEndButton/ScrollToEndButton.js]: Fixed in latest commit.

File: CHANGELOG.md (hunk 4 above)
Comment (deepalic) [reply CHANGELOG.md]: Fixed in latest commit.

**Comment by OEvgeny** [comment]
Closed in favour of #5251
